src/product/index.ts

```diff
@@ -0,0 +1,76 @@
+import api from "../api";
+import { Product, Response } from "../types";
+import { hasError } from "../util";
+import { DataTransform } from 'node-json-transform'
+import { productTransformRule } from "../mappings/product";
+
+export async function fetchProducts(params: any): Promise<any | Response> {
+  let response = {} as Product[] | Response
+
+  const payload = {
+    "json": {
+      "params": {
+        "rows": params.viewSize,
+        "start": params.viewIndex
+      },
+      "query": "*:*",
+      "filter": `docType: PRODUCT`
+    } as any
+  }
+
+  const queryFields = params.queryFields ? params.queryFields : "productId productName internalName";
+
+  // checking that if the params has filters, and then adding the filter values in the payload filter
+  // for each key present in the params filters
+  if (params.filters) {
+    Object.keys(params.filters).map((key: any) => {
+
+      if (Array.isArray(params.filters[key])) {
+        // TODO: need to check how to handle the condition when we want to do AND on values
+        payload.json.filter += ` AND ${key}: (${params.filters[key].join(' OR ')})`
```

Comment:
We could make this operator configurable

Reply:
Sure Sir, will handle the case by expecting that each filter will have an operator and a value with associated with it.
```js
{
  "productId": {
    "value": ['10000', '10002'],
    "op": 'AND'
  }
}
```